classes/class.ilOpencastPageComponentPluginGUI.php

```diff
@@ -124,32 +124,81 @@ public function __construct()
         parent::__construct();
     }
 
+    private function ensureContext(string $command): void
+    {
+        $this->ensureParameters();
+        // we connot rely on ilCtrl->getCmdClass() here, because it is seems to be "overwritten" by the pageeditor which is really bad practice.
+        $current_comand_class = strtolower(
+            (string) ($this->dic->http()->request()->getQueryParams()['cmdClass'] ?? null)
+        );
+        if ($current_comand_class !== strtolower(self::class)) {
+            // we are not in the right context, redirect to insert
+            $this->redirect($command, $command);
+            return;
```

Comment:
Do we still need a return after the redirect?

Reply:
not technically, but for readability i think it's good to see directly that there is an ‘end’ to this method